sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/StreamingForeachBatchHelper.scala

```diff
@@ -135,7 +136,10 @@ object StreamingForeachBatchHelper extends Logging {
       sessionHolder: SessionHolder): (ForeachBatchFnType, AutoCloseable) = {
 
     val port = SparkConnectService.localPort
-    val connectUrl = s"sc://localhost:$port/;user_id=${sessionHolder.userId}"
+    var connectUrl = s"sc://localhost:$port/;user_id=${sessionHolder.userId}"
+    Connect.getAuthenticateToken.foreach { token =>
```

Comment:
Nit: If you like reusing code then factor this into a helper function in `sql/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala`.

```scala
def authenticationTokenParam: String = getAuthenticateToken.map(token => ";token=" + token).getOrElse("")
```